### ISSUE TITLE
Support for Alternate Dependencies

### ISSUE THREAD

**Issue by halostatue** [DannyBen/bashly#354]
### Description

I want to be able to specify alternate dependencies so that this dependency comment can be true:

```yml
dependencies:
  jq: |-
    Ensure that you have installed jq or gojq. If gojq is available, it will
    be preferred.

    https://stedolan.github.io/jq/download/
```

This could be supported in either array or dict configurations by adding support for a dict value:

```yml
dependencies:
  - { commands: [jq, gojq], message: "Ensure you have installed either jq or gojq." }
```

or

```yml
dependencies:
  jq:
    commands: [jq, gojq]
    message: "Ensure that you have installed either jq or gojq."
```

This would be a minimal change to the generated code, because `builtin command` already accepts multiple arguments and returns 0 if any one of them is found.

**Comment by DannyBen** [comment]
Yes. I can see the value.

The proposed hash syntax is not bad, and the fact that `command -v {{ dependency }}` will work with multiple arguments helps.

I wonder if there is also any need to provide some indication to the bash script of the found dependencies. This could justify the fact that this hash has a key. For example, provide the bash script with `$dependencies['jq']='/usr/bin/gojq'` (the result of the `command -v` maybe)

**Comment by DannyBen** [comment]
For reference, this is the spot:

https://github.com/DannyBen/bashly/blob/b5c3ec6ea6cec19834cdb8edb644deb46fc3b4ce/lib/bashly/views/command/dependencies_filter.gtx#L4-L13

**Comment by DannyBen** [comment]
> I was looking at what the change would be and it appears that you transform dependencies: [jq, aws] into dependencies: { jq: nil, aws: nil }, so it would be possible to provide $dependencies for all specified dependencies.

I am not following.

**Comment by DannyBen** [comment]
I am not doing any transformation of dependencies. Since this property allows either an array of commands or a hash of command => help pairs, you may see transformation happen on its own. 

In any case, the code above is too complex, especially compared to the fact that the [`Command`](https://github.com/DannyBen/bashly/blob/master/lib/bashly/script/command.rb) class which is the owner of the dependencies, currently has zero lines of code that handle it. It is just a simple property.

Now I see that adding this option, will add a **third** syntax for the dependencies:

- array
- hash of strings
- hash of hashes with command and message keys

I will take a deeper look into it tomorrow.

**Comment by DannyBen** [comment]
See my implementation. Note that it does not yet include the bash `dependencies` array - this will be done separately, since the PR is large enough.

**Comment by halostatue** [comment]
That looks really clean and makes sense. If I’m reading your version correctly, you can mix hash styles:

```yaml
dependencies:
  aws: Install aws-cli
  jq:
    command: [gojq, jq]
    help: Install gojq or jq
```

If that’s the case, it will be very useful (I had explicitly decided against that in my implementation because the logic of validation was getting too complex; extracting it out into a class makes the validation much simpler).

**Comment by DannyBen** [comment]
Yes, you can.

I am now changing it slightly to even allow:

```yaml
dependencies:
  git: ~
  http_client:
    command: [curl, wget]
```

and, I am working on adding the `dependencies` variable to the generated script + to have it display in `inspect_args`.

I wanted your opinion on something:

Do you think this variable should be called `dependencies` or `deps`?
Consider the fact that it is in the `inspect_args`, where we have `args`, `flags`, `other_args`

```
# inpect_args output
args: none

dependencies:
- ${dependencies[git]} = /usr/bin/git
- ${dependencies[http_client]} = /usr/bin/curl
```

**Comment by halostatue** [comment]
`deps` is fine, I think.

**Comment by DannyBen** [comment]
I think its done.

**Comment by DannyBen** [comment]
Released in 1.0.1, thanks for the feature suggestions.
